src/pages/workspaces/workspace/tools/LaunchAnalysisModal.js

```diff
@@ -58,27 +60,44 @@ export default class LaunchAnalysisModal extends Component {
   }
 
   componentDidMount() {
-    const { workspaceId: { namespace, name }, config: { rootEntityType } } = this.props
+    const { workspaceId: { namespace, name } } = this.props
+    const { entityType } = this.state
 
     Rawls.workspace(namespace, name).entityMetadata().then(
-      entityMetadata => {
-        const { attributeNames, idName } = entityMetadata[rootEntityType]
-        this.setState({ attributeNames, idName })
-      },
+      entityMetadata => this.setState({ entityMetadata, entitySetExists: !!entityMetadata[`${entityType}_set`] }),
```

Comment:
It looks like `entitySetExists` can be computed from existing props and state, so it doesn't need to be stored separately as state.